src/main/java/unrc/dose/UserProfileEndpoint.java

```diff
@@ -0,0 +1,93 @@
+
+    /** The logger. */
+    static final Logger LOGEER = LoggerFactory.getLogger(UserProfileEndpoint.class);
+
+    /** main namespace of this endpoint. */
+    private static final String NAME_SPACE = "/users";
+    
+  
+    
+    @Override
+    public void bind(final SparkSwagger restApi) {
+        
+        restApi.endpoint(
+            endpointPath(NAME_SPACE)
+            .withDescription(
+                "UserProfile REST API exposing all UserProfile utilities"
+                ),
+            (q, a) -> LOGEER.info("Logging Received request for UserProfile Rest API"))
+        
+           .post(
+                   path("/:id/profile")
+                       .withDescription("creates a new user profile")
+                        .withQueryParam()
+                            .withName("displayName")
+                            .withDescription("user profile display name").and()
+                       .withQueryParam()                            
+                       .withName("twitterId")
+                       .withDescription("user profile twitter id").and()
+                      
+                        .withResponseType(String.class),
+                    (req, res) -> {
+                    	if (!User.exists(Integer.parseInt(req.params("id")))) {
+           					throw new UserNotFoundException(String.valueOf(Integer.parseInt(req.params("id"))));
+          					}
+                    	
+                        return UserProfileService.userprofile(
+                        Integer.parseInt(req.params("id")),req.queryParams("displayName"),req.queryParams("twitterId"));
+                    	
+                 }
+                )
+             .put(
+                   path("/:id/profile")
+                       .withDescription("Modify user profile")
+                       .withQueryParam()
+                       .withName("displayName")
+                       .withDescription("user profile display name").and()
+                       .withQueryParam()
+                    	.withName("twitterId")
+                       	.withDescription("user profile twitter id").and()  
+                        .withResponseType(String.class),
+                     
+                    (req, res) -> {
+                    	
+                    	    return UserProfileService.updateUp(Integer.parseInt(req.params("id")), 
+                    	   		req.queryParams("displayName"),req.queryParams("twitterId"));
+          					
+                            }
+                )
+                .get(
+                       path("/:id/profile")
+                            .withDescription("Will return the user profile of user's id")
+                             
+                            .withResponseType(String.class),
+                        (req, res) -> {
+                        	
+                        	return UserProfileService.searchById(Integer.parseInt(req.params("id")));
```

Comment:
The returned value should be a JSON representation of a UserProfile, but the methods name suggests that this only returns a UserProfile (if found, what happens if there is no UserProfile or User?)